src/software/ai/hl/stp/tactic/goalie/goalie_tactic.cpp

```diff
@@ -0,0 +1,63 @@
+#include "software/ai/hl/stp/tactic/goalie/goalie_tactic.h"
+
+#include "software/ai/evaluation/calc_best_shot.h"
+#include "software/ai/hl/stp/action/stop_action.h"  // TODO (#1888): remove this dependency
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/point.h"
+
+GoalieTactic::GoalieTactic(std::shared_ptr<const GoalieTacticConfig> goalie_tactic_config)
+    : Tactic(true,
+             {RobotCapability::Move, RobotCapability::Dribble, RobotCapability::Chip}),
+      fsm(),
+      goalie_tactic_config(goalie_tactic_config),
+      control_params{
+          GoalieFSM::ControlParams{.goalie_tactic_config = goalie_tactic_config}}
+{
+}
+
+void GoalieTactic::updateWorldParams(const World &world) {}
+
+void GoalieTactic::updateControlParams(
+    std::shared_ptr<const GoalieTacticConfig> updated_goalie_tactic_config)
+{
+    // Update the control parameters stored by this Tactic
+    control_params.goalie_tactic_config = updated_goalie_tactic_config;
+}
+
+double GoalieTactic::calculateRobotCost(const Robot &robot, const World &world) const
+{
+    // We don't prefer any particular robot to be the goalie, as there should only
+    // ever be one robot that can act as the goalie
+    return 0.5;
+}
+
+void GoalieTactic::calculateNextAction(ActionCoroutine::push_type &yield)
+{
+    auto stop_action = std::make_shared<StopAction>(false);
```

Comment:
What happened to the previous move_action and chip_action that was also declared here in the old `goalie_tactic.cpp`

Reply:
Those were used in the old implementation, but now that everything has been migrated to goalie_fsm.h, we don't need those in this file anymore